### PR TITLE
Allow formatting with 'x' token

### PR DESCRIPTION
**Pull request by systemcatch**
- [x] 🧪 Added **tests** for changed code.
- [x] 🛠️ All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [ ] 📚 Updated **documentation** for changed code.
- [x] ⏩ Code is **up-to-date** with the `master` branch.

There's a failing test which we need to discuss, otherwise this is working ok.

```shell
(arrow) chris@ThinkPad:~/arrow$ python
Python 3.7.4 (default, Sep 19 2019, 11:01:37) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import arrow
>>> adt=arrow.utcnow()
>>> adt.format("x")
'1585669870688329'
>>> adt.format("X")
'1585669870'
```

### CODE REVIEW

**Comment by systemcatch** [comment]
All tests passing is weird as that test I've added specifically fails on my local machine. :thinking:

**Comment by systemcatch** [comment]
This is the test failure I get.
```
====================================== FAILURES ======================================
__________________ TestDateTimeFormatterFormatToken.test_timestamp ___________________

self = <tests.test_formatter.TestDateTimeFormatterFormatToken object at 0x7f3af38b2f90>

    def test_timestamp(self):
    
        timestamp = time.time()
        dt = datetime.utcfromtimestamp(timestamp)
    
        assert self.formatter._format_token(dt, "X") == str(int(timestamp))
    
        # time.time() may return a float with greater than 6 digits of precision
        # time_parts = str(timestamp).split(".")
        # if len(time_parts[1]) > 6:
        #     shortened = time_parts[1][:6]
        #     truncated_timestamp = time_parts[0] + "." + shortened
        #     timestamp = float(truncated_timestamp)
        #
        # assert self.formatter._format_token(dt, "x") == str(int(timestamp*1000000))
    
        # TODO failing test for discussion with Jad
        bad_timestamp = 1585668666.7198436
        dt = datetime.utcfromtimestamp(bad_timestamp)
>       assert self.formatter._format_token(dt, "x") == str(
            int(bad_timestamp * 1000000)
        )
E       AssertionError: assert '1585668666719844' == '1585668666719843'
E         - 1585668666719844
E         ?                ^
E         + 1585668666719843
E         ?                ^

tests/test_formatter.py:125: AssertionError
```

**Comment by jadchaar** [comment]
That is peculiar that CI is passing, but it is failing on your machine. This must be a discrepancy regarding how floating point numbers or integers are represented on different OSes or hardware configurations.

This seems to be occurring because the cast to an int is truncating the `7198436/1e6` into `719843` rather than the rounded version of `719844`. You can probably solve this by rounding the division to 6 digits like this: `ts = calendar.timegm(dt.utctimetuple()) + round(dt.microsecond / 1000000, 6)`.

My only worry about this is that it technically goes against the IEEE-754 rounding we use in the parser.

File: tests/test_formatter.py [hunk 1]
```diff
@@ -107,8 +107,13 @@ def test_timestamp(self):
 
         timestamp = time.time()
         dt = datetime.utcfromtimestamp(timestamp)
+
```
Comment (jadchaar) [inline tests/test_formatter.py:110]: Was this extra newline intentional?

File: tests/test_formatter.py [hunk 2]
```diff
@@ -107,8 +107,13 @@ def test_timestamp(self):
...
         assert self.formatter._format_token(dt, "X") == str(int(timestamp))
 
+        # time.time() may return a float with greater than 6 digits of precision
+        rounded_ts = str(round(timestamp * 1000000))
+        assert self.formatter._format_token(dt, "x") == rounded_ts.format("{f}")
```
Comment (jadchaar) [inline tests/test_formatter.py:115]: Any reason you chose to do `self.formatter._format_token(dt, "x") == rounded_ts.format("{f}")` over just `self.formatter._format_token(dt, "x") == rounded_ts`?

File: arrow/formatter.py [hunk 3]
```diff
@@ -100,6 +100,10 @@ def _format_token(self, dt, token):
         if token == "X":
             return str(calendar.timegm(dt.utctimetuple()))
 
+        if token == "x":
+            ts = calendar.timegm(dt.utctimetuple()) + (dt.microsecond / 1000000)
+            return str(int(ts * 1000000))
```
Comment (jadchaar) [inline arrow/formatter.py:105]: Do we have to worry about rounding >6 digits of precision here as well?

File: tests/test_formatter.py (hunk 1 above)
Comment (systemcatch) [reply tests/test_formatter.py:110]: Nope, removing

File: tests/test_formatter.py (hunk 2 above)
Comment (systemcatch) [reply tests/test_formatter.py:115]: Without that format it defaults to an exponential style which causes an AssertionError.

```
E       AssertionError: assert '1585764864000000' == '1.5857648648e+15'
E         - 1585764864000000
E         + 1.5857648648e+15
```

File: arrow/formatter.py (hunk 3 above)
Comment (systemcatch) [reply arrow/formatter.py:105]: Not here since `datetime` microseconds is capped at 6 digits.

File: tests/test_formatter.py (hunk 2 above)
Comment (jadchaar) [reply tests/test_formatter.py:115]: Oh interesting. Did not realize that :D.

File: arrow/formatter.py (hunk 3 above)
Comment (jadchaar) [reply arrow/formatter.py:105]: Let's put a test in our suite to test this edge case.

Comment (jadchaar) [reply arrow/formatter.py:105]: Ah in that case I don't think a test is necessary then.

**Comment by systemcatch** [comment]
Do you think we need any docs changes for this?

**Comment by jadchaar** [comment]
> Do you think we need any docs changes for this?

Good question. I think it may have been covered when we added the parsing support: https://arrow.readthedocs.io/en/latest/#supported-tokens.
